### PR TITLE
Clean up OpenSearch connector docs

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Follow up to discussion with @wendigo on https://github.com/trinodb/trino/pull/20257

Apart from commented changes everything is just reformatting, rewording, and clean up.

I can send a follow up PR for the Elasticsearch connector docs as well.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/connector/opensearch.md [hunk 1]
```diff
@@ -311,60 +371,21 @@ It is not allowed to use `asRawJson` and `isArray` flags simultaneously for the
...
-| \_score  | The document score returned by the OpenSearch query |
-| \_source | The source of the original document                 |
-
-## Predicate push down
-
-The connector supports [predicate push down](predicate-pushdown) for the following data types:
-
-| OpenSearch   | Trino         | Supported     |
-|--------------|---------------|---------------|
-| `binary`     | `VARBINARY`   | `NO`          |
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/opensearch.md:326]: I dropped the ones with NO from the table and added a sentence after the table that nothing else is supported.

File: docs/src/main/sphinx/connector/opensearch.md [hunk 2]
```diff
@@ -424,6 +446,49 @@ FROM
 ```{include} query-table-function-ordering.fragment
 ```
 
+## Performance
+
+The connector includes a number of performance improvements, detailed in the
+following sections.
+
+### Parallel data access
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/opensearch.md:454]: This is new

File: docs/src/main/sphinx/connector/opensearch.md [hunk 3]
```diff
@@ -19,12 +21,12 @@ properties as appropriate for your setup:
 
 ```text
 connector.name=opensearch
-opensearch.host=localhost
+opensearch.host=es.example.com
```
Comment (wendigo) [inline docs/src/main/sphinx/connector/opensearch.md:24]: os.example.com?

Comment (mosabua) [reply docs/src/main/sphinx/connector/opensearch.md:24]: Good point .. silly me

Comment (mosabua) [reply docs/src/main/sphinx/connector/opensearch.md]: used search.example....

**Review by martint — APPROVED** [review]

**Pull request merged**